src/pollen.py

```diff
@@ -72,3 +72,21 @@ def stop(self):
         cache = snap.SnapCache()
         pollen = cache[SNAP_NAME]
         pollen.stop()
+
+    def check_rng_file(self, charm_state):
+        """Check if the rng-tools-debian file needs modification.
+
+        Args:
+            charm_state: Pollen charm's CharmState instance.
+        """
+        file = Path("/etc/default/rng-tools-debian")
+        # The file already has this line commented by default,
+        # so we should check if it appears in the file twice (commented and actually written).
+        # The file needs the commented code header so the rngd service does not fail.
+        if (
+            'RNGDOPTIONS="--fill-watermark=90% --feed-interval=1"' in charm_state.rng_tools_file
+            and '# RNGDOPTIONS="--fill-watermark=90% --feed-interval=1"'
+            not in charm_state.rng_tools_file
+        ):
+            return
+        file.write_text('RNGDOPTIONS="--fill-watermark=90% --feed-interval=1"', encoding="utf-8")
```

Comment:
You probably need to update the state from here too, even if that's not really used later, better have it in sync